comms/dht/src/test_utils/makers.rs

```diff
@@ -123,9 +124,9 @@ pub fn make_valid_message_signature(node_identity: &NodeIdentity, message: &[u8]
         .to_encoded_bytes()
 }
 
-pub fn make_dht_inbound_message(
+pub fn make_dht_inbound_message<T: prost::Message>(
     node_identity: &NodeIdentity,
-    body: Vec<u8>,
+    body: &T,
```

Comment:
You might be able to save a lot of `&[...].to_vec()` calls with this type of thing:
```rust
fn foo<R: Debug + ?Sized, T: AsRef<R>>(t: T)  {
    println!("{:?}", t.as_ref());
}

fn main() {
    let v = vec![1];
    foo::<Vec<i32>, _>(&v);
    foo::<[i32], _>(&[1,2,3]);
}
```

Reply:
We want this to work for any impl prost::Message, [u8] does not implement this but Vec<u8> does which is why, for tests only, we get the wrong looking &[..].to_vec()